src/main/java/frc/team2412/robot/subsystems/VisionSubsystem.java

```diff
@@ -0,0 +1,128 @@
+package frc.team2412.robot.subsystems;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.networktables.NetworkTableEvent;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.Hardware;
+import io.github.oblarg.oblog.annotations.Log;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import org.photonvision.PhotonCamera;
+import org.photonvision.targeting.PhotonPipelineResult;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+public class VisionSubsystem extends SubsystemBase {
+	private PhotonCamera photonCamera;
+	private PhotonPipelineResult latestResult;
+	private BiConsumer<Pose2d, Double> poseConsumer;
+	/** Null if no known robot pose, otherwise the last calculated robot pose from vision data. */
+	private Pose3d robotPose = null;
```

Comment:
It'd be helpful to know the timestamp of the last pose as well (ideally in FPGA time)

Reply:
When would we need the timestamp outside of updating the pose estimator? (Also the timestamp is in the same epoch as `getFPGATimestamp()`, so it would be in FPGA time)

Reply:
all done